securedrop/source_app/main.py

```diff
@@ -113,27 +120,40 @@ def submit():
                   "error")
             return redirect(url_for('main.lookup'))
 
-        fnames = []
         journalist_filename = g.source.journalist_filename
         first_submission = g.source.interaction_count == 0
 
         if msg:
             g.source.interaction_count += 1
-            fnames.append(
+            fname = (
                 store.save_message_submission(
                     g.filesystem_id,
                     g.source.interaction_count,
                     journalist_filename,
                     msg))
+            submission = Submission(g.source, fname)
+            db_session.add(submission)
+            db_session.commit()
+
         if fh:
-            g.source.interaction_count += 1
-            fnames.append(
-                store.save_file_submission(
-                    g.filesystem_id,
-                    g.source.interaction_count,
-                    journalist_filename,
-                    fh.filename,
-                    fh.stream))
+            for file_submitted in fh:
+                g.source.interaction_count += 1
+                fname = (
+                    store.save_file_submission(
+                        g.filesystem_id,
+                        g.source.interaction_count,
+                        journalist_filename,
+                        file_submitted.filename,
+                        file_submitted.stream))
+                submission = Submission(g.source, fname)
+                db_session.add(submission)
+                db_session.commit()
+
+                filename = Markup.escape(file_submitted.filename)
+                html_msg = render_template(
+                    'next_submission_flashed_message.html',
+                    html_contents="{0}".format(filename))
+                flash(Markup(html_msg), "info")
```

Comment:
So the flashed messages look a bit wonky:

![screen shot 2017-11-14 at 4 14 42 pm](https://user-images.githubusercontent.com/7832803/32811916-22ea0d54-c957-11e7-8777-20ccbf1fc13f.png)

Primarily because the styling on the background of the new flashed messages is not green, and because the new flashed messages consume a lot of space on the screen. An alternative approach (that is simpler design wise) could be creating a list of files successfully uploaded as they complete and printing "Files x, y, and z successfully uploaded" in the existing success box.

Reply:
I actually removed the green background from the filenames in the flash messages. We can have a css class to make them look decent. 
I will have to check how to keep flashing messages in the same success box.